src/System.Private.CoreLib/shared/System/BitOps.cs

```diff
@@ -2,37 +2,641 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System.Diagnostics;
 using System.Runtime.CompilerServices;
 using System.Runtime.InteropServices;
 using System.Runtime.Intrinsics.X86;
-
+using System.Threading;
 using Internal.Runtime.CompilerServices;
 
+// Some routines inspired by the Stanford Bit Widdling Hacks by Sean Eron Anderson:
+// http://graphics.stanford.edu/~seander/bithacks.html
+
 namespace System
 {
+    // This class is not meant to be merged in its current form.
+    // The duplicate methods are intentional - they show the pervasiveness of similar (known) code across the stack
+    // If this PR progresses, the duplicates will be consolidated and optimized, and the call sites (corert, coreclr, corefx) updated.
```

Comment:
I think this comment is out-of date now. I think it would also be good to just update the existing call sites to call the consolidated methods (and to remove the methods that were replaced).

Reply:
Though I can make the changes, pretending that the push has happened. It won't compile but it will show the value.

Reply:
> But updating the callsites is not straightforward - they are in corert, coreclr and corefx as well as ancilliary assemblies. I can do the coreclr ones but will need to wait for the auto-push on the others I think?

As long as the sources live in CoreCLR, it should be fine (and things will flow to the other repos naturally, especially for things in the `shared` folder). For anything where the source lives elsewhere (such as in CoreFX or CoreRT), they would need to be updated separately after this PR is merged.

Reply:
OK, CoreCLR changes done